nixos/neovim/default.nix

```
@@ -0,0 +1,182 @@
+          description = "Extra build depends to add `LIBRARY_PATH` and `CPATH`.";
+        };
+      };
+    };
+  };
+  config =
+    let
+      # Inspired from https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/programs/nix-ld.nix
+      build-dependent-pkgs = with pkgs;
+        [
+          zlib
+          zstd
+          stdenv.cc.cc
+          curl
+          openssl
+          attr
+          libssh
+          bzip2
+          libxml2
+          acl
+          libsodium
+          util-linux
+          xz
+          systemd
+          # Packages not included in `nix-ld`'s NixOSModule
+          glib
+          libcxx
+        ]
+        ++ cfg.extraDependentPackages;
+
+      makePkgConfigPath = x: makeSearchPathOutput "dev" "lib/pkgconfig" x;
+      makeIncludePath = x: makeSearchPathOutput "dev" "include" x;
+
+      nvim-depends-library = pkgs.buildEnv {
+        name = "nvim-depends-library";
+        paths = map lib.getLib build-dependent-pkgs;
+        extraPrefix = "/lib/nvim-depends";
+        pathsToLink = [ "/lib" ];
+        ignoreCollisions = true;
+      };
+      nvim-depends-include = pkgs.buildEnv {
+        name = "nvim-depends-include";
+        paths = splitString ":" (makeIncludePath build-dependent-pkgs);
+        extraPrefix = "/lib/nvim-depends/include";
+        ignoreCollisions = true;
+      };
+      nvim-depends-pkgconfig = pkgs.buildEnv {
+        name = "nvim-depends-pkgconfig";
+        paths = splitString ":" (makePkgConfigPath build-dependent-pkgs);
+        extraPrefix = "/lib/nvim-depends/pkgconfig";
+        ignoreCollisions = true;
+      };
+      buildEnv = [
+        "CPATH=${config.home.profileDirectory}/lib/nvim-depends/include"
+        "CPLUS_INCLUDE_PATH=${config.home.profileDirectory}/lib/nvim-depends/include/c++/v1"
+        "LD_LIBRARY_PATH=${config.home.profileDirectory}/lib/nvim-depends/lib"
+        "LIBRARY_PATH=${config.home.profileDirectory}/lib/nvim-depends/lib"
+        "NIX_LD_LIBRARY_PATH=${config.home.profileDirectory}/lib/nvim-depends/lib"
+        "PKG_CONFIG_PATH=${config.home.profileDirectory}/lib/nvim-depends/pkgconfig"
+      ];
+    in
+    mkIf cfg.enable
+      {
+        xdg.configFile = {
+          "nvim/init.lua".source = ../../init.lua;
+          "nvim/lua".source = ../../lua;
+          "nvim/snips".source = ../../snips;
+          "nvim/tutor".source = ../../tutor;
+        };
+        home.packages = with pkgs; [
+          ripgrep
+        ] ++ optionals cfg.setBuildEnv [ patchelf nvim-depends-library nvim-depends-include nvim-depends-pkgconfig ];
+        home.extraOutputsToInstall = optional cfg.setBuildEnv "nvim-depends";
+        home.shellAliases.nvim = optionalString cfg.setBuildEnv (concatStringsSep " " buildEnv) + " SQLITE_CLIB_PATH=${pkgs.sqlite.out}/lib/libsqlite3.so " + "nvim";
+
+        programs.neovim = {
+          enable = true;
+          viAlias = true; # Replace from vi&vim to neovim
```

This is just a preference, but I still prefer vi -> vim, and vim -> neovim
```suggestion
          viAlias = true; # Replace from vi&vim to neovim
```

---

ah. It was my preferred setting.
As for Alias, I think it would be better if the user could select it, so I will remove it.

---

Ah probably we can comment this out and leave it in such that ppl now that these exist.

Sometimes nix documentation can be hard to navigate 😢